src/fmain.pas

```diff
@@ -6171,6 +6159,40 @@ procedure TfrmMain.UpdateSelectedDrives;
   UpdateDriveButtonSelection(btnRightDrive, FrameRight);
 end;
 
+function TfrmMain.GenerateTitle: String;
+var ServernameString: String;
+  begin
+    ServernameString := '';
+    if Length(UniqueInstance.ServernameByUser) > 0 then
+      ServernameString := ' [' + UniqueInstance.ServernameByUser + ']';
+
+    Result := Format('%s%s %s build %s; %s',
+        ['Double Commander',
+        ServernameString,
+        dcVersion,
+        dcRevision,
+        dcBuildDate]
+    );
+  end;
+
+procedure TfrmMain.UpdateMainTitleBar;
+var sTmp: String;
+begin
+    if (fspDirectAccess in ActiveFrame.FileSource.Properties) and (gShowCurDirTitleBar = true) then
```

Comment:
I think it is better to use "if gShowCurDirTitleBar and (fspDirectAccess in ActiveFrame.FileSource.Properties) then"

Reply:
fixed